src/transformers/modeling_t5.py

```diff
@@ -916,3 +916,22 @@ def forward(self, **kwargs):
             ) + decoder_outputs  # TODO(thom): Add z_loss https://github.com/tensorflow/mesh/blob/fa19d69eafc9a482aff0b59ddd96b025c0cb207d/mesh_tensorflow/layers.py#L666
 
         return decoder_outputs + encoder_outputs
+
+    def prepare_inputs_for_generation(self, input_ids, past, attention_mask, **kwargs):
+        assert past is not None, "past has to be defined for encoder_outputs"
+
+        # first step
+        if type(past) is tuple:
```

Comment:
I would rather keep it as it is because `step` is not really a model input and the function should prepare the model inputs. Agree though that the `type(past) is tuple` statement has room for improval!